app/src/main/java/chat/rocket/android/chatrooms/presentation/ChatRoomsPresenter.kt

```diff
@@ -245,4 +249,32 @@ class ChatRoomsPresenter @Inject constructor(private val view: ChatRoomsView,
     fun disconnect() {
         client.disconnect()
     }
+
+    /**
+     * Logout from current server.
+     */
+    fun logout() {
+        launchUI(strategy) {
+            try {
+                clearTokens()
+                client.logout()
+                //TODO: Add the code to unsubscribe to all subscriptions.
+                client.disconnect()
+                view.onLogOut()
+            } catch (e: Exception) {
```

Comment:
What about catching `RocketChatException` instead of `Exception`?

Reply:
Makes sense. Will change :)